src/prefect/task_engine.py

```diff
@@ -598,9 +599,6 @@ def example_flow():
                         msg=msg,
                     )
 
-                    self._is_started = False
-                    self._client = None
```

Comment:
Not sure yet if this has unintended consequences 👀 

Passes quite a bit of the test suite fine. With this, when attempting to access `TaskEngine.client` outside of this context manager, a `RuntimeError("Engine has not started.")` is thrown which makes sense if/when the run has yet to start, but in my case here I'm attempting to access it _after_ the run has started, but also after it has completed in order to roll it back. This is because transaction hooks execute _outside_ the task run engine/context. 

So I figured I could either extend this context to include the execution of the transaction hooks or do what I'm doing here which is to allow access back to the task's client after the task run has completed. 

The former seemed like a much more substantial change though. I also like that this implementation maintains some of the separation between the Transaction interface and the TaskEngine.

Reply:
Agree strongly with @desertaxle here - once the engine is shut down, it's not guaranteed to be in a healthy state that can be relied upon for making additional updates to the task run (hence the `RuntimeError`).